clang/test/CheckedC/inferred-bounds/widened-bounds.c

```diff
@@ -0,0 +1,154 @@
+// Tests for datafow analysis for bounds widening of _Nt_array_ptr's.
+//
+// RUN: %clang_cc1 -fdump-widened-bounds %s 2>1 | FileCheck %s
+
+_Nt_array_ptr<char> p : count(0);
+_Nt_array_ptr<char> q : count(0);
+
+void f1() {
+  if (*p) {}
+
+// CHECK: In function: f1
+// CHECK:  [B1]
+// CHECK: upper_bound(p) = 1
+}
+
+void f2() {
+  int a;
+  if (*p) {
```

Comment:
This isn't correct, the bounds aren't widened by subsequent tests of (*p).    You would want the code to have the form
```
if (*p)   // bounds are now count(1)
 if (*(p + 1))   // bounds are now count(2)
   if (*(p + 2)) // bounds are now count(3)